diff --git a/lib/core/media_player/media_player.dart b/lib/core/media_player/media_player.dart
index dea59f4a..2bc96321 100644
--- a/lib/core/media_player/media_player.dart
+++ b/lib/core/media_player/media_player.dart
@@ -10,7 +10,6 @@ import 'package:harmonoid/core/configuration/configuration.dart';
 import 'package:harmonoid/core/filesystem_media_library.dart';
 import 'package:harmonoid/core/media_player/base_media_player.dart';
 import 'package:harmonoid/core/media_player/mixin/audio_service_mixin.dart';
-import 'package:harmonoid/core/media_player/mixin/discord_rpc_mixin.dart';
 import 'package:harmonoid/core/media_player/mixin/history_playlist_mixin.dart';
 import 'package:harmonoid/core/media_player/mixin/lastfm_mixin.dart';
 import 'package:harmonoid/core/media_player/mixin/mpris_mixin.dart';
@@ -40,9 +39,7 @@ import 'package:harmonoid/utils/constants.dart';
 /// Implementation to handle the media playback & other related functionalities.
 ///
 /// {@endtemplate}
-class MediaPlayer extends ChangeNotifier
-    with AudioServiceMixin, DiscordRpcMixin, HistoryPlaylistMixin, LastFmMixin, MprisMixin, SystemMediaTransportControlsMixin, WindowsTaskbarMixin
-    implements BaseMediaPlayer {
+class MediaPlayer extends ChangeNotifier with AudioServiceMixin, HistoryPlaylistMixin, LastFmMixin, MprisMixin, SystemMediaTransportControlsMixin, WindowsTaskbarMixin implements BaseMediaPlayer {
   static const Duration kDefaultCrossfadeDuration = Duration(seconds: 5);
   static const Duration kMinCrossfadeDuration = Duration(seconds: 2);
   static const Duration kMaxCrossfadeDuration = Duration(seconds: 30);
@@ -66,7 +63,6 @@ class MediaPlayer extends ChangeNotifier
     await Future.wait(
       [
         ensureInitializedAudioService(),
-        ensureInitializedDiscordRpc(),
         ensureInitializedHistoryPlaylist(),
         ensureInitializedLastFm(),
         ensureInitializedMpris(),
@@ -441,7 +437,6 @@ class MediaPlayer extends ChangeNotifier
     _player.dispose();
     _tagReader.dispose();
     disposeAudioService();
-    disposeDiscordRpc();
     disposeHistoryPlaylist();
     disposeLastFm();
     disposeMpris();
diff --git a/lib/core/media_player/mixin/discord_rpc_mixin.dart b/lib/core/media_player/mixin/discord_rpc_mixin.dart
deleted file mode 100644
index 009750ef..00000000
--- a/lib/core/media_player/mixin/discord_rpc_mixin.dart
+++ /dev/null
@@ -1,153 +0,0 @@
-// ignore_for_file: implementation_imports
-
-import 'dart:async';
-import 'dart:io';
-import 'package:flutter_discord_rpc/flutter_discord_rpc.dart';
-import 'package:flutter/rendering.dart';
-import 'package:image/image.dart' as img;
-import 'package:safe_local_storage/file_system.dart';
-import 'package:synchronized/synchronized.dart';
-import 'package:media_kit/src/player/native/utils/temp_file.dart';
-
-import 'package:harmonoid/api/activity_set.dart';
-import 'package:harmonoid/core/configuration/configuration.dart';
-import 'package:harmonoid/core/media_player/base_media_player.dart';
-import 'package:harmonoid/extensions/string.dart';
-import 'package:harmonoid/localization/localization.dart';
-import 'package:harmonoid/models/playable.dart';
-import 'package:harmonoid/utils/async_file_image.dart';
-import 'package:harmonoid/utils/rendering.dart';
-
-/// {@template discord_rpc_mixin}
-///
-/// DiscordRpcMixin
-/// ---------------
-/// package:flutter_discord_rpc mixin for [BaseMediaPlayer].
-///
-/// {@endtemplate}
-mixin DiscordRpcMixin implements BaseMediaPlayer {
-  static const String kApplicationId = '881480706545573918';
-  static const String kDefaultLargeImage = 'cover_default';
-  static const String kPauseSmallImage = 'pause';
-  static const String kPlaySmallImage = 'play';
-
-  static bool get supported => (Platform.isLinux || Platform.isMacOS || Platform.isWindows) && Configuration.instance.discordRpc;
-
-  Future<void> ensureInitializedDiscordRpc() async {
-    if (!supported) return;
-
-    try {
-      await FlutterDiscordRPC.initialize(kApplicationId);
-      final instance = FlutterDiscordRPC.instance..connect();
-
-      _instanceDiscordRpc = instance;
-
-      addListener(_listenerDiscordRpc);
-    } catch (exception, stacktrace) {
-      debugPrint(exception.toString());
-      debugPrint(stacktrace.toString());
-    }
-  }
-
-  Future<void> disposeDiscordRpc() async {
-    if (!supported) return;
-    _instanceDiscordRpc = null;
-  }
-
-  void resetFlagsDiscordRpc() {
-    _flagPlayableDiscordRpc = null;
-  }
-
-  void _listenerDiscordRpc() {
-    if (_instanceDiscordRpc?.isConnected != true) return;
-
-    _lockDiscordRpc.synchronized(() async {
-      bool notify = false;
-
-      if (_flagPlayableDiscordRpc != current) {
-        _flagPlayableDiscordRpc = current;
-        notify = true;
-        try {
-          final image = cover(uri: current.uri);
-          _largeImageDiscordRpc = switch (image) {
-            AsyncFileImage() => await _getFileUrl(current, image.getFile()),
-            FileImage() => await _getFileUrl(current, image.file),
-            NetworkImage() => image.url,
-            _ => null,
-          }!;
-        } catch (_) {
-          _largeImageDiscordRpc = kDefaultLargeImage;
-        }
-      }
-      if (_flagPlayingDiscordRpc != state.playing) {
-        _flagPlayingDiscordRpc = state.playing;
-        notify = true;
-      }
-      if (((_flagPositionDiscordRpc ?? Duration.zero) - state.position).abs() > const Duration(seconds: 10)) {
-        _flagPositionDiscordRpc = state.position;
-        notify = true;
-      }
-
-      if (notify) {
-        await _instanceDiscordRpc?.setActivity(
-          activity: RPCActivity(
-            state: current.subtitle.take(2).join(', ').ellipsis(128).nullIfBlank(),
-            details: current.title.ellipsis(128).nullIfBlank(),
-            timestamps: state.playing
-                ? RPCTimestamps(
-                    start: DateTime.now().subtract(state.position).millisecondsSinceEpoch,
-                    end: DateTime.now().subtract(state.position).add(state.duration).millisecondsSinceEpoch,
-                  )
-                : null,
-            assets: RPCAssets(
-              largeImage: _largeImageDiscordRpc,
-              smallImage: state.playing ? kPlaySmallImage : kPauseSmallImage,
-              largeText: current.description.join(' • ').ellipsis(128).nullIfBlank(),
-              smallText: state.playing ? Localization.instance.PLAYING : Localization.instance.PAUSED,
-            ),
-            buttons: [
-              RPCButton(
-                label: Localization.instance.FIND,
-                url: 'https://www.google.com/search?q=${Uri.encodeComponent([current.title, ...current.subtitle.take(2)].where((e) => e.isNotEmpty).join(' '))}',
-              ),
-            ],
-            activityType: ActivityType.listening,
-          ),
-        );
-      }
-    });
-  }
-
-  Future<String?> _getFileUrl(Playable playable, FutureOr<File?> inputFuture) async {
-    final input = await inputFuture;
-    final output = await TempFile.create();
-
-    if (input == null) return null;
-
-    try {
-      final deviceId = Configuration.instance.identifier;
-      final activitySet = ActivitySet();
-
-      final cmd = img.Command()
-        ..decodeImageFile(input.path)
-        ..copyResize(width: 256)
-        ..encodeJpg(quality: 85)
-        ..writeToFile(output.path);
-
-      await cmd.executeThread();
-
-      return await activitySet(deviceId, playable, output);
-    } finally {
-      await output.delete_();
-    }
-  }
-
-  FlutterDiscordRPC? _instanceDiscordRpc;
-  final Lock _lockDiscordRpc = Lock();
-
-  Playable? _flagPlayableDiscordRpc;
-  bool? _flagPlayingDiscordRpc;
-  Duration? _flagPositionDiscordRpc;
-
-  String? _largeImageDiscordRpc;
-}
diff --git a/lib/utils/actions.dart b/lib/utils/actions.dart
index 694aaa20..cb4a99f0 100644
--- a/lib/utils/actions.dart
+++ b/lib/utils/actions.dart
@@ -87,7 +87,6 @@ void mediaPlayerUpdateCurrentOnUpdateCurrent(String uri) {
     AsyncFileImage.reset(uri);
     MediaPlayer.instance
       ..resetFlagsAudioService()
-      ..resetFlagsDiscordRpc()
       ..resetFlagsMpris()
       ..resetFlagsSystemMediaTransportControls();
     NowPlayingColorPaletteNotifier.instance.resetCurrent();
diff --git a/lib/utils/rendering.dart b/lib/utils/rendering.dart
index e8a76ccc..0b42061d 100644
--- a/lib/utils/rendering.dart
+++ b/lib/utils/rendering.dart
@@ -219,7 +219,6 @@ ImageProvider cover({
       if (MediaPlayer.instance.current.uri == result) {
         MediaPlayer.instance
           ..resetFlagsAudioService()
-          ..resetFlagsDiscordRpc()
           ..resetFlagsMpris()
           ..resetFlagsSystemMediaTransportControls();
         NowPlayingColorPaletteNotifier.instance.resetCurrent();
diff --git a/pubspec.yaml b/pubspec.yaml
index fb5392ab..ea26dc18 100644
--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -23,7 +23,6 @@ dependencies:
   ffi: ^2.1.2
   file_picker: ^10.3.3
   fluentui_system_icons: ^1.1.136
-  flutter_discord_rpc: 1.0.0
   flutter_displaymode: ^0.7.0
   flutter_local_notifications:
     git:
       url: https://github.com/alexmercerind/flutter_local_notifications.git
       ref: e8763a2
       path: flutter_local_notifications
