diff --git a/lib/core/media_player/media_player_mixin_registry.dart b/lib/core/media_player/media_player_mixin_registry.dart
--- a/lib/core/media_player/media_player_mixin_registry.dart
+++ b/lib/core/media_player/media_player_mixin_registry.dart
@@ -4,7 +4,6 @@
 import 'package:harmonoid/core/media_player/media_player.dart';
 import 'package:harmonoid/core/media_player/mixin/audio_service_mixin.dart';
 import 'package:harmonoid/core/media_player/mixin/audio_session_mixin.dart';
-import 'package:harmonoid/core/media_player/mixin/discord_rpc_mixin.dart';
 import 'package:harmonoid/core/media_player/mixin/history_playlist_mixin.dart';
 import 'package:harmonoid/core/media_player/mixin/lastfm_mixin.dart';
 import 'package:harmonoid/core/media_player/mixin/media_player_mixin.dart';
@@ -25,7 +24,6 @@ final class MediaPlayerMixinRegistry {
       [
         if (AudioServiceMixin.supported) AudioServiceMixin(player),
         if (AudioSessionMixin.supported) AudioSessionMixin(player),
-        if (DiscordRpcMixin.supported) DiscordRpcMixin(player),
         if (HistoryPlaylistMixin.supported) HistoryPlaylistMixin(player),
         if (LastFmMixin.supported) LastFmMixin(player),
         if (MprisMixin.supported) MprisMixin(player),
diff --git a/pubspec.yaml b/pubspec.yaml
--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -25,7 +25,6 @@ dependencies:
   # https://github.com/miguelpruivo/flutter_file_picker/pull/1964
   file_picker: ^12.0.0-beta.7
   fluentui_system_icons: ^1.1.136
-  flutter_discord_rpc: 1.0.0
   flutter_displaymode: ^0.7.0
   flutter_local_notifications:
     git:
